tools/make_topog/topog.c

```diff
@@ -497,13 +498,24 @@ void create_realistic_topog(int nx_dst, int ny_dst, const double *x_dst, const d
 	mask_src[i] = 1.0;
     }
   }
+  if(on_grid) {
```

Comment:
@ThomasNeumann2, looks good though I think lines 402 to here could be replaced by something slightly simpler.  The relevant info that comes out of those lines are the source grid indices that indicate where it overlaps with the dest grid.  Since the two grids are the same, indices from the domain struct could probably be used.

Reply:
@nikizadehgfdl : Our use case is the Baltic Sea. The models have a hor. grid resolution between 1n.m. and 3n.m. The model topography of the Baltic requires some manual modifications, small islands, channels, coast line, sills etc. A new topog run removes e.g. some islands. We found that the regridding function is doing it. Our solution is to skip this function if appropriate and maybe someone else could also made use of it.
@mlee03 : That is probably true. For me it was simpler to add the on_grid case and don't touch any other code. Thus, everything remains the same when using make_topog.